packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -1817,7 +1831,7 @@ function performWork(minExpirationTime: ExpirationTime, dl: Deadline | null) {
   findHighestPriorityRoot();
 
   if (enableProfilerTimer) {
-    resumeActualRenderTimerIfPaused();
+    resumeActualRenderTimerIfPaused(minExpirationTime === Sync);
```

Comment:
Letting the profiler timer know when it's processing sync work enables it to deduct that time from pending async work (by adding it onto the total pause time later). This ensures we don't count time spent processing a batch commit against yielded async work.